### PR TITLE
Update Flow to 0.84

### PR DESCRIPTION
**Pull request by gaearon**
Updating to v0.84 first is a first step that avoids breaking changes in subsequent versions of Flow. v0.84 includes the new inexact object syntax, which is necessary to align with fbsource and unblock RN sync fix. Afterwards we can continue updating Flow in increments.

### CODE REVIEW

**Comment by necolas** [comment]
Why upgrade to version 0.84 specifically?

**Comment by gaearon** [comment]
They said it should be easy compared to some big breaking change after. It also gives us new inexact object syntax, which is necessary to align with fbsource and unblock RN sync fix.

**Comment by gaearon** [comment]
We should upgrade to a later version too for sure but I started with this as MVP.

File: yarn.lock [hunk 1]
```diff
@@ -4864,6 +4864,7 @@ eslint-plugin-no-unsafe-innerhtml@1.0.16:
 
 "eslint-plugin-react-internal@link:./scripts/eslint-rules":
   version "0.0.0"
+  uid ""
```
Comment (TrySound) [inline yarn.lock]: Is this an accident?

Comment (gaearon) [reply yarn.lock]: I just ran Yarn. No idea.

**Pull request merged**